keanu-python/tests/test_vertex_operations.py

```diff
@@ -135,126 +138,168 @@ def test_can_do_less_than_or_equal_to_with_vertex_on_rhs(lhs: Union[Vertex, nump
 
-# yapf: disable
-@pytest.mark.parametrize("lhs, rhs, expected_result", [
-    (Const(np.array([10., 20.])), Const(np.array([1., 2.])), np.array([11, 22])),
-    (Const(np.array([10., 20.])),       np.array([1., 2.]) , np.array([11, 22])),
-    (Const(np.array([10., 20.])),                     2.   , np.array([12, 22])),
-])
-# yapf: enable
-def test_can_do_addition(lhs: Union[Vertex], rhs: Union[Vertex, numpy_types, float],
-                         expected_result: numpy_types) -> None:
-    result: Union[Vertex, np._ArrayLike[Any]] = lhs + rhs
-    assert isinstance(result, Vertex)
-    assert (result.get_value() == expected_result).all()
-    result = rhs + lhs
-    assert isinstance(result, Vertex)
-    assert (result.get_value() == expected_result).all()
+arithmetic_op_value_types = Union['np._ArrayLike[int]', int]
 
 
-# yapf: disable
-@pytest.mark.parametrize("lhs, rhs, expected_result", [
-    (Const(np.array([10., 20.])), Const(np.array([1., 2.])), np.array([9, 18])),
-    (Const(np.array([10., 20.])),       np.array([1., 2.]) , np.array([9, 18])),
-    (Const(np.array([10., 20.])),                     2.   , np.array([8, 18])),
-])
-# yapf: enable
-def test_can_do_subtraction(lhs: Vertex, rhs: Union[Vertex, numpy_types, float], expected_result: numpy_types) -> None:
-    result: Union[Vertex, np._ArrayLike[Any]] = lhs - rhs
+def infer_result_type(lhs: Union[Vertex, numpy_types, int], rhs: Union[Vertex, numpy_types, int]) -> Type[vertex_types]:
+    if is_floating_type(lhs) or is_floating_type(rhs):
+        return Double
+    else:
+        return Integer
+
+
+def result_should_be_vertex(lhs: Union[Vertex, numpy_types], rhs: Union[Vertex, numpy_types, float]) -> bool:
+    if isinstance(lhs, Vertex) or isinstance(rhs, Vertex):
+        return True
+    else:
+        return False
+
+
+def assert_is_correct_vertex_type_and_expected_value(lhs: Union[Vertex, numpy_types, int],
+                                                     rhs: Union[Vertex, numpy_types, int], result: Any,
+                                                     expected_result: arithmetic_op_value_types) -> None:
     assert isinstance(result, Vertex)
+    assert type(result) == infer_result_type(lhs, rhs)
     assert (result.get_value() == expected_result).all()
-    result = rhs - lhs
-    assert isinstance(result, Vertex)
-    assert (result.get_value() == -expected_result).all()
 
 
 # yapf: disable
+@pytest.mark.parametrize("lhs_constructor", types_to_test)
+@pytest.mark.parametrize("rhs_constructor", types_to_test)
 @pytest.mark.parametrize("lhs, rhs, expected_result", [
-    (Const(np.array([3., 2.])), Const(np.array([5., 7.])), np.array([15, 14])),
-    (Const(np.array([3., 2.])),       np.array([5., 7.]) , np.array([15, 14])),
-    (Const(np.array([3., 2.])),                 5.       , np.array([15, 10])),
+    (         np.array([10, 20]), np.array([1, 2]) , np.array([11, 22])),
```

Comment:
Shouldn't we be creating different combinations of lhs and rhs with `integer` and `double`? Looks like we're always doing integer here

Reply:
Oh that's done in:

```
@pytest.mark.parametrize("lhs_constructor", types_to_test)
@pytest.mark.parametrize("rhs_constructor", types_to_test)
```